### PR TITLE
CaseContact status enum changes

### PR DESCRIPTION
**Pull request by thejonroberts**
### What github issue is this PR for, if any?

No issue: syntax changes only. None of this is necessary.

While working on CaseContacts, I noticed the status enum was defined oddly. I started poking around. The enum is used to define `.active?`/`.started?` and not much else - doesn't use any of the other methods or scopes that come along with an enum definition. It is also a string column, which is atypical. That is for easy compatibility with the wicked wizard form steps - but standard integer column would make for faster queries.

### What changed, and _why_?

The main thing I wanted to do is define the enum in a way that makes sense. But then I kept going to see if we could better use the enum, and set things up for a transition to integer column. That won't be necessary until performance is a real-world concern. It would require migrating existing data, which could be delicate. But these changes would enable it. The definition & wizard form updates would work if the enum column was changed to an integer type.

- Explicitly define the enum (_If this is all you want to do here, I'm happy with that_)
- Use the methods and scopes that are defined by doing so
- Integrate those changes into the wizard step updates
- Integrate the enum into factories

### How is this **tested**? (please write tests!) 💖💪
Current specs. ~~Could add/modify specs for each of the enum-created scopes, will probably look at that a bit, especially if people want to do this.~~ I think the scope changes in specs are enough?

### CODE REVIEW

File: app/models/case_contact.rb [hunk 1]
```diff
@@ -40,16 +40,28 @@ class CaseContact < ApplicationRecord
   after_save_commit ::CaseContactMetadataCallback.new
 
   # Corresponds to the steps in the controller, so validations for certain columns can happen at those steps.
-  # These steps must be listed in order and have an html template in case_contacts/form.
+  # These steps must be listed in order, have an html template in case_contacts/form, & be listed in the status enum
   FORM_STEPS = %i[details notes expenses].freeze
-  enum status: (%w[started active] + FORM_STEPS.map(&:to_s)).zip((%w[started active] + FORM_STEPS.map(&:to_s))).to_h
```
Comment (thejonroberts) [inline app/models/case_contact.rb:45]: This was written this way to allow adding wizard form steps and have them automatically incorporated into the enum. I don't think that is worth it being so hard to read. I would rather define explicitly. It evaluates to:
```ruby
status: {
  "started"=>"started", 
  "active"=>"active", 
  "details"=>"details", 
  "notes"=>"notes", 
  "expenses"=>"expenses"
}
```

File: spec/factories/case_contacts.rb [hunk 2]
```diff
@@ -1,16 +1,18 @@
 FactoryBot.define do
+  # note: uses enum status column, see:
+  # https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#enum-traits
```
Comment (elasticspoon) [inline spec/factories/case_contacts.rb:3]: I found this a bit hard to follow initially. It made sense that if there is an enum then traits are automagically created for it but I did not realize that `active`, `started` within the factories were calls to tho traits. 

I would appreciate like 2 sentences explaining that because I don't think is a pattern used elsewhere in the project.

This was the piece of context I was missing https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#traits-within-traits.

**Review by elasticspoon — APPROVED** [review]
LGTM.

File: spec/factories/case_contacts.rb (hunk 2 above)
Comment (thejonroberts) [reply spec/factories/case_contacts.rb:3]: That makes sense, will explain better!

Comment (thejonroberts) [reply spec/factories/case_contacts.rb]: updated.
